### DISCUSSION TITLE
[BUG]: the `DevToolsPlugin` with Edge as `preferredBrowser` does not seem to work

### DISCUSSION THREAD

**Discussion by Adam-it** [dotnet/dev-proxy#1230] [Q&A] [answered]
### Description

So I configured the Dev Proxy to use `DevToolsPlugin` with Edge but when I start the devproxy I get the browser and dev tools but seems there is some error present in the devtools console and nothing is logged in the chrome debugger.

![image](https://github.com/microsoft/dev-proxy/assets/58668583/5e894167-7edf-4aa8-9c11-2ae247d475db)

on the console, the devproxy logs the requests

![image](https://github.com/microsoft/dev-proxy/assets/58668583/4ed96315-464a-40eb-9150-3b9dfe72f18c)

but in the network tab of the devtools I don't get anything 
![image](https://github.com/microsoft/dev-proxy/assets/58668583/ed95660f-1f2f-48e5-bef0-06526cffd88d)

### Expected behaviour

I should see 'catched' requests in the Edge DevTools in Network tab

### Actual behaviour

I don't get anything in the Edge DevTools in Network tab and I see some errors on console tab

### Steps to reproduce

1. install dev proxy
2. configure `DevToolsPlugin` following the guide
3. start `devproxy --no-mocks --log-level Info --failure-rate 0`

### Dev Proxy Version

0.13.0

### Operating system (environment)

Windows

### Shell

PowerShell

### Configuration file

{
  "plugins": [
    {
      "name": "DevToolsPlugin",
      "enabled": false,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "configSection": "devTools"
    },
    {
      "name": "LatencyPlugin",
      "enabled": false,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "configSection": "latencyPlugin"
    },
    {
      "name": "RetryAfterPlugin",
      "enabled": true,
      "pluginPath": "plugins\\dev-proxy-plugins.dll"
    },
    {
      "name": "GraphSelectGuidancePlugin",
      "enabled": true,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "urlsToWatch": [
        "https://graph.microsoft.com/v1.0/*",
        "https://graph.microsoft.com/beta/*",
        "https://graph.microsoft.us/v1.0/*",
        "https://graph.microsoft.us/beta/*",
        "https://dod-graph.microsoft.us/v1.0/*",
        "https://dod-graph.microsoft.us/beta/*",
        "https://microsoftgraph.chinacloudapi.cn/v1.0/*",
        "https://microsoftgraph.chinacloudapi.cn/beta/*"
      ]
    },
    {
      "name": "ODSPSearchGuidancePlugin",
      "enabled": true,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "urlsToWatch": [
        "https://graph.microsoft.com/v1.0/*",
        "https://graph.microsoft.com/beta/*",
        "https://graph.microsoft.us/v1.0/*",
        "https://graph.microsoft.us/beta/*",
        "https://dod-graph.microsoft.us/v1.0/*",
        "https://dod-graph.microsoft.us/beta/*",
        "https://microsoftgraph.chinacloudapi.cn/v1.0/*",
        "https://microsoftgraph.chinacloudapi.cn/beta/*"
      ]
    },
    {
      "name": "GraphBetaSupportGuidancePlugin",
      "enabled": true,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "urlsToWatch": [
        "https://graph.microsoft.com/beta/*",
        "https://graph.microsoft.us/beta/*",
        "https://dod-graph.microsoft.us/beta/*",
        "https://microsoftgraph.chinacloudapi.cn/beta/*"
      ]
    },
    {
      "name": "GraphSdkGuidancePlugin",
      "enabled": true,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "urlsToWatch": [
        "https://graph.microsoft.com/v1.0/*",
        "https://graph.microsoft.com/beta/*",
        "https://graph.microsoft.us/v1.0/*",
        "https://graph.microsoft.us/beta/*",
        "https://dod-graph.microsoft.us/v1.0/*",
        "https://dod-graph.microsoft.us/beta/*",
        "https://microsoftgraph.chinacloudapi.cn/v1.0/*",
        "https://microsoftgraph.chinacloudapi.cn/beta/*"
      ]
    },
    {
      "name": "ODataPagingGuidancePlugin",
      "enabled": true,
      "pluginPath": "plugins\\dev-proxy-plugins.dll"
    },
    {
      "name": "GraphClientRequestIdGuidancePlugin",
      "enabled": true,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "urlsToWatch": [
        "https://graph.microsoft.com/v1.0/*",
        "https://graph.microsoft.com/beta/*",
        "https://graph.microsoft.us/v1.0/*",
        "https://graph.microsoft.us/beta/*",
        "https://dod-graph.microsoft.us/v1.0/*",
        "https://dod-graph.microsoft.us/beta/*",
        "https://microsoftgraph.chinacloudapi.cn/v1.0/*",
        "https://microsoftgraph.chinacloudapi.cn/beta/*"
      ]
    },
    {
      "name": "CachingGuidancePlugin",
      "enabled": true,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "configSection": "cachingGuidance"
    },
    {
      "name": "RateLimitingPlugin",
      "enabled": false,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "configSection": "rateLimiting"
    },
    {
      "name": "MockResponsePlugin",
      "enabled": false,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "configSection": "mocksPlugin"
    },
    {
      "name": "GraphMockResponsePlugin",
      "enabled": true,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "configSection": "mocksPlugin"
    },
    {
      "name": "GraphRandomErrorPlugin",
      "enabled": true,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "configSection": "graphRandomErrorsPlugin"
    },
    {
      "name": "ExecutionSummaryPlugin",
      "enabled": false,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "configSection": "executionSummaryPlugin"
    },
    {
      "name": "MinimalPermissionsPlugin",
      "enabled": true,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "configSection": "minimalPermissionsPlugin"
    },
    {
      "name": "MinimalPermissionsGuidancePlugin",
      "enabled": false,
      "pluginPath": "plugins\\dev-proxy-plugins.dll"
    },
    {
      "name": "MockGeneratorPlugin",
      "enabled": false,
      "pluginPath": "plugins\\dev-proxy-plugins.dll"
    },
    {
      "name": "DevToolsPlugin",
      "enabled": true,
      "pluginPath": "plugins\\dev-proxy-plugins.dll",
      "configSection": "devTools"
    }
  ],
  "devTools": {
    "preferredBrowser": "Edge"
  },
  "urlsToWatch": [
    "https://graph.microsoft.com/v1.0/*",
    "https://graph.microsoft.com/beta/*",
    "https://graph.microsoft.us/v1.0/*",
    "https://graph.microsoft.us/beta/*",
    "https://dod-graph.microsoft.us/v1.0/*",
    "https://dod-graph.microsoft.us/beta/*",
    "https://microsoftgraph.chinacloudapi.cn/v1.0/*",
    "https://microsoftgraph.chinacloudapi.cn/beta/*",
    "https://*.sharepoint.*/*_api/*",
    "https://*.sharepoint.*/*_vti_bin/*",
    "https://*.sharepoint-df.*/*_api/*",
    "https://*.sharepoint-df.*/*_vti_bin/*"
  ],
  "mocksPlugin": {
    "mocksFile": "responses.json"
  },
  "graphRandomErrorsPlugin": {
    "allowedErrors": [ 429, 500, 502, 503, 504, 507 ]
  },
  "executionSummaryPlugin": {
    "groupBy": "url"
  },
  "minimalPermissionsPlugin": {
    "type": "delegated"
  },
  "cachingGuidance": {
    "cacheThresholdSeconds": 5
  },
  "latencyPlugin": {
    "minMs": 200,
    "maxMs": 10000
  },
  "rateLimiting": {
    "costPerRequest": 2,
    "rateLimit": 120,
    "retryAfterSeconds": 5
  },
  "rate": 50,
  "labelMode": "text",
  "logLevel": "info"
}

**Comment by garrytrinder** [comment]
Apologies for the trouble @Adam-it

Can you share the version number of Edge you are using?

Is it possible for you to try the latest `EdgeDev` and see if you have the same issue?

**Comment by Adam-it** [comment]
> Apologies for the trouble @Adam-it
> 
> Can you share the version number of Edge you are using?
> 
> Is it possible for you to try the latest `EdgeDev` and see if you have the same issue?

Will check and come back with additional I go over the night.
Yes I will check the latest edgeDev as well 👍

**Comment by Adam-it** [comment]
@garrytrinder so the edge I am using is 
![image](https://github.com/microsoft/dev-proxy/assets/58668583/b12af4ad-41c2-47aa-bfd1-6606e8c3ef14)

however, I installed the latest EdgeDev and in this case, it works like a charm 🤩
![image](https://github.com/microsoft/dev-proxy/assets/58668583/2589a5a4-1876-436b-8cf2-5cda7736f556)

**Comment by garrytrinder** [comment]
Thanks for the update @Adam-it

We will investigate and see if we can figure out what is going wrong on standard Edge.

@waldekmastykarz are you able to reproduce on macOS?

**Comment by waldekmastykarz** [comment]
Have you disabled the startup boost in Edge @Adam-it (https://learn.microsoft.com/en-us/microsoft-cloud/dev/dev-proxy/technical-reference/devtoolsplugin#known-issues)? Also, do you have all Edge windows closed before starting proxy?

**Comment by Adam-it** [comment]
> Have you disabled the startup boost in Edge @Adam-it (https://learn.microsoft.com/en-us/microsoft-cloud/dev/dev-proxy/technical-reference/devtoolsplugin#known-issues)? Also, do you have all Edge windows closed before starting proxy?

Yes and no.
I will close all edge instances and recheck and let you know

**Comment by Adam-it** [comment]
@waldekmastykarz having all the Edge windows closed and then starting the proxy did not bring any difference and I still get the same errors
![image](https://github.com/microsoft/dev-proxy/assets/58668583/d93d87e5-ec94-4a61-95f9-be9b41c904c8)

**Comment by waldekmastykarz** [comment]
Thank you for confirming. I'll try to repro asap and will let you know if I need any more info to fix it. Sorry for the trouble

**Comment by Adam-it** [comment]
> Thank you for confirming. I'll try to repro asap and will let you know if I need any more info to fix it. Sorry for the trouble

no problem at all. It's working with the EdgeDev so it's not a blocker for me 🙂

**Comment by waldekmastykarz** [comment]
When you refer to Edge in your config, we're trying to run https://github.com/microsoft/dev-proxy/blob/df28158bb1aea4d96653db95c39f190478aa75a4/dev-proxy-plugins/Inspection/DevToolsPlugin.cs#L74. I wonder if Edge for Business is not a different version of Edge that we're not supporting at the moment? Could you double check if the installation directory matches what we've got?

**Comment by garrytrinder** [comment]
I'm experiencing a different problem when using `Edge`.

Screenshots are taken from the browser opened by Proxy. Note that no `for Business` is shown in the About screen.

![image](https://github.com/microsoft/dev-proxy/assets/11563347/e74d0079-d48c-48c7-9151-767c319d6575)

![image](https://github.com/microsoft/dev-proxy/assets/11563347/6385f2bc-859f-4b1d-a707-c15e5450c632)

![image](https://github.com/microsoft/dev-proxy/assets/11563347/a04dca13-75d0-4041-bc95-13cf6a29896d)

Edge Dev works fine.

Install location for Edge is 

```
C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe
```

If I open the About page after opening Edge myself, you see the same version but has `for Business` added to the name.

![image](https://github.com/microsoft/dev-proxy/assets/11563347/61929180-240f-45ec-ae31-c629a26d3e00)

**Comment by waldekmastykarz** [comment, marked as answer]
@Adam-it it seems like you've opened dev tools for dev tools. If you look at your original window, it points to dev tools but the screen is empty meaning Edge failed to load dev tools. Try closing all Edge processes using Task Explorer and restarting proxy with dev tools pointing at Edge, and then expand dev tools from the side.

**Comment by Adam-it** [comment]
> When you refer to Edge in your config, we're trying to run
> 
> https://github.com/microsoft/dev-proxy/blob/df28158bb1aea4d96653db95c39f190478aa75a4/dev-proxy-plugins/Inspection/DevToolsPlugin.cs#L74
> 
> . I wonder if Edge for Business is not a different version of Edge that we're not supporting at the moment? Could you double check if the installation directory matches what we've got?

yep it seems exactly the same
![image](https://github.com/microsoft/dev-proxy/assets/58668583/fef44580-deeb-4da9-87bf-e8cc90b91901)

**Comment by Adam-it** [comment]
> @Adam-it it seems like you've opened dev tools for dev tools. If you look at your original window, it points to dev tools but the screen is empty meaning Edge failed to load dev tools. Try closing all Edge processes using Task Explorer and restarting proxy with dev tools pointing at Edge, and then expand dev tools from the side.

ok perfect. I did what you wrote here and works. Edge started with dev tools 👍
![image](https://github.com/microsoft/dev-proxy/assets/58668583/2fbfe0b8-238a-4c2e-aafe-16cafdcbdff1)

We may close the issue. Sorry for the trouble 👍

**Comment by waldekmastykarz** [comment]
No need to be sorry. I appreciate you double checking it with us. If anything else, perhaps we should clarify it some more in our docs. Thank you for bringing it up
